gremlin-go/driver/traversal.go

```diff
@@ -96,3 +82,31 @@ func (t *Traversal) Iterate() (*Traversal, <-chan bool, error) {
 
 	return t, r, nil
 }
+
+func (t *Traversal) HasNext() (bool, error) {
+	results, err := t.getResults()
+	if err != nil {
+		return false, err
+	}
+	return !(*results).IsEmpty(), nil
```

Comment:
Why do you need to dereference the results here?

Reply:
He needs to because this is a pointer to an interface (equivalent to double pointer)

Reply:
Should I convert this to a direct reference to an interface instead?